### PR TITLE
Fix searchpath in the template lookup

### PR DESCRIPTION
**Pull request by mkrizek**
##### SUMMARY

Based on https://github.com/ansible/ansible/pull/49819#discussion_r242965155

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

template lookup

### CODE REVIEW

File: lib/ansible/plugins/lookup/template.py [hunk 1]
```diff
@@ -76,19 +76,18 @@ def run(self, terms, variables, **kwargs):
...
-                        newsearchpath = []
-                        for p in searchpath:
-                            newsearchpath.append(os.path.join(p, 'templates'))
-                            newsearchpath.append(p)
-                        searchpath = newsearchpath
-                    else:
-                        searchpath = [self._loader._basedir, os.path.dirname(lookupfile)]
+                    # set Jinja2 internal search path for includes
+                    searchpath = variables.get('ansible_search_path', [])
+                    searchpath.extend([self._loader.get_basedir(), os.path.dirname(lookupfile)])
```
Comment (bcoca) [inline lib/ansible/plugins/lookup/template.py]: get_basedir should not be needed, the ansible_search_path should include the correct one

Comment (bcoca) [reply lib/ansible/plugins/lookup/template.py]: the lookupfile should not presume a templates/ subdir so we may want to add it always after appending it, not before

**Review by bcoca — CHANGES_REQUESTED** [review]

File: lib/ansible/plugins/lookup/template.py (hunk 1 above)
Comment (mkrizek) [reply lib/ansible/plugins/lookup/template.py]: Thanks, fixed.

File: lib/ansible/plugins/lookup/template.py [hunk 2]
```diff
@@ -87,8 +87,8 @@ def run(self, terms, variables, **kwargs):
                             newsearchpath.append(os.path.join(p, 'templates'))
                             newsearchpath.append(p)
                         searchpath = newsearchpath
-                    else:
-                        searchpath = [self._loader._basedir, os.path.dirname(lookupfile)]
+                    searchpath.append(os.path.dirname(lookupfile))
```
Comment (bcoca) [inline lib/ansible/plugins/lookup/template.py]: this should be prepended, as the list controls the priority in which the search happens, the template itself should be the first location scanned

**Review by bcoca — CHANGES_REQUESTED** [review]

**Review by bcoca — APPROVED** [review]

**Pull request merged**